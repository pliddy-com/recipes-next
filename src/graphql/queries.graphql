# Query for Homepage RecipeCollection (all recipes)

query homePageQuery {
  recipeCollection(order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe collection based on filter

query recipeCollection($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    total
    items {
      ...RecipeDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)

query listPageQuery($where: TagFilter) {
  tagCollection(where: $where, limit: 1) {
    total
    items {
      slug
      title
      linkedFrom {
        recipeCollection(limit: 24) {
          total
          items {
            ...RecipeSummary
          }
        }
      }
    }
  }
}

# Query for recipe slugs

query recipeSlugsCollection($where: RecipeFilter) {
  recipeCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query categorySlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query tagSlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
      linkedFrom {
        recipeCollection {
          total
        }
      }
    }
  }
}

# query for Taxonomy with Category & Subcategory levels

query taxonomyCollection($where: TaxonomyFilter) {
  taxonomyCollection(where: $where, limit: 1) {
    items {
      ...TaxonomyDefault

      tag {
        ...TagDefault
      }

      childrenCollection(limit: 24) {
        total
        items {
          ... on Tag {
            ...TagDefault

            linkedFrom {
              recipeCollection(limit: 1) {
                total
              }
            }
          }
          ...TaxonomyWithChildren
        }
      }
    }
  }
}
