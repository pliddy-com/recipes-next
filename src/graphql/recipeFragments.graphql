# Fragments with default fields for Recipe-specific sections:
#   * Ingredients
#   * Instructions
#   * Image
#   * Recipe

fragment IngredientsDefault on IngredientSection {
  sys {
    id
  }
  title
  slug
  label
  ingredientList
}

fragment InstructionsDefault on InstructionSection {
  sys {
    id
  }
  title
  slug
  label
  instructionList
}

fragment ImageDefault on Asset {
  sys {
    id
  }
  __typename
  title
  description
  contentType
  fileName
  size
  url
  height
  width
}

fragment DescriptionDefault on Recipe {
  description {
    json
    links {
      entries {
        inline {
          __typename
          sys {
            id
            __typename
          }
        }
        hyperlink {
          __typename
          sys {
            id
            __typename
          }
        }
        block {
          __typename

          sys {
            id
            __typename
          }
        }
      }
      assets {
        hyperlink {
          __typename
          title
        }
        block {
          __typename
          title
        }
      }
    }
  }
}

fragment RecipeDefault on Recipe {
  sys {
    id
  }
  __typename
  title
  slug
  description {
    json
  }
  abstract
  image {
    ...ImageDefault
  }
  ingredientsCollection(limit: 10) {
    items {
      ...IngredientsDefault
    }
  }
  equipment
  instructionsCollection(limit: 10) {
    items {
      ...InstructionsDefault
    }
  }
  notes
  tagsCollection(limit: 10) {
    items {
      ...TagDefault
    }
  }
}
