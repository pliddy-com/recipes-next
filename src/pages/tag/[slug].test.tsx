// import testing-library methods
import { render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import TagPage from './[slug]';

import { ListPageItemFragment } from 'types/generated/graphql';

describe('TagPage in tag/[slug].tsx', () => {
  describe('when there is page content', () => {
    it('it renders the TagPage if there is content', () => {
      const pageContent = {
        slug: 'baking',
        title: 'Baking',
        linkedFrom: {
          recipeCollection: {
            total: 3,
            items: [
              {
                sys: {
                  id: '13DABWfxYIdpP3YGrXS6kk',
                  __typename: 'Sys',
                },
                __typename: 'Recipe',
                title: 'Biscuits',
                slug: 'biscuits',
                abstract:
                  'Traditional baking powder biscuits, baked on a parchment-lined baking sheet.',
                image: {
                  sys: {
                    id: 'B6C4D23gXNDNfGVrxb1DK',
                    __typename: 'Sys',
                  },
                  __typename: 'Asset',
                  title: 'Biscuits',
                  description:
                    'A batch of baking soda biscuits on a parchment-lined baking sheet.',
                  contentType: 'image/jpeg',
                  fileName: 'biscuits.jpg',
                  size: 2583242,
                  url: 'https://images.ctfassets.net/fo9qwg6zarbt/B6C4D23gXNDNfGVrxb1DK/9d8e62c1faf660f7e542eed14e044b57/biscuits.jpg',
                  height: 3024,
                  width: 4032,
                },
                tagsCollection: {
                  items: [
                    {
                      sys: {
                        id: '014itYSAM0Vg2zypswRokk',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Baking',
                      slug: 'baking',
                    },
                    {
                      sys: {
                        id: '30UFYfPfMSgZ4d1kPmeCPM',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Biscuits',
                      slug: 'biscuits',
                    },
                    {
                      sys: {
                        id: '3AQGtSkvlxQgQQMwJqaglw',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Southern',
                      slug: 'southern',
                    },
                  ],
                  __typename: 'RecipeTagsCollection',
                },
              },
              {
                sys: {
                  id: '5qvltoWoIIZvzBJVMWoZ85',
                  __typename: 'Sys',
                },
                __typename: 'Recipe',
                title: 'No-Knead Bread',
                slug: 'no-knead-bread',
                abstract:
                  'A no-knead bread recipe using instant (bread machine) yeast and a dutch oven or covered ceramic bread pan.',
                image: {
                  sys: {
                    id: '6HZDSm8AK4iQHr9v2UOGpq',
                    __typename: 'Sys',
                  },
                  __typename: 'Asset',
                  title: 'No Knead Bread',
                  description:
                    'Loaf of no-knead bread cooling on a wire baking rack.',
                  contentType: 'image/jpeg',
                  fileName: 'no-knead-bread.JPG',
                  size: 2943075,
                  url: 'https://images.ctfassets.net/fo9qwg6zarbt/6HZDSm8AK4iQHr9v2UOGpq/17569d5be00441216338c7800e5ebc2c/no-knead-bread.JPG',
                  height: 3024,
                  width: 4032,
                },
                tagsCollection: {
                  items: [
                    {
                      sys: {
                        id: '014itYSAM0Vg2zypswRokk',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Baking',
                      slug: 'baking',
                    },
                    {
                      sys: {
                        id: '5oUMcctJ6A9EzMnLS4XQdO',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Bread',
                      slug: 'bread',
                    },
                  ],
                  __typename: 'RecipeTagsCollection',
                },
              },
              {
                sys: {
                  id: '3cbUDWABNsvaBc1OcuHzNw',
                  __typename: 'Sys',
                },
                __typename: 'Recipe',
                title: 'No-Knead Olive Bread',
                slug: 'no-knead-olive-bread',
                abstract: 'No-knead bread with olives and seasoning.',
                image: {
                  sys: {
                    id: '2rUa4IhfkLtRRBmHS6ihcD',
                    __typename: 'Sys',
                  },
                  __typename: 'Asset',
                  title: 'Olive Bread',
                  description: 'Loaf of no-knead olive bread',
                  contentType: 'image/jpeg',
                  fileName: 'olive-bread.JPG',
                  size: 2849709,
                  url: 'https://images.ctfassets.net/fo9qwg6zarbt/2rUa4IhfkLtRRBmHS6ihcD/e953337e00d1eba9551b15e7eee18229/olive-bread.JPG',
                  height: 3024,
                  width: 4032,
                },
                tagsCollection: {
                  items: [
                    {
                      sys: {
                        id: '014itYSAM0Vg2zypswRokk',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Baking',
                      slug: 'baking',
                    },
                    {
                      sys: {
                        id: '5oUMcctJ6A9EzMnLS4XQdO',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Bread',
                      slug: 'bread',
                    },
                    {
                      sys: {
                        id: '6cWaSfUY3oU1oKERx29iAk',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Olives',
                      slug: 'olives',
                    },
                    {
                      sys: {
                        id: '58gioVdtuUwFvnPamjosAv',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Mediterranean',
                      slug: 'mediterranean',
                    },
                  ],
                  __typename: 'RecipeTagsCollection',
                },
              },
            ],
            __typename: 'RecipeCollection',
          },
          __typename: 'TagLinkingCollections',
        },
        __typename: 'Tag',
      };

      const { container } = render(
        <TagPage
          pageContent={pageContent as ListPageItemFragment}
          preview={false}
        />
      );

      // assert that page head tags are rendered
      const titleTag = document.getElementsByTagName('title')[0];
      expect(titleTag).toBeInTheDocument();
      expect(titleTag.text.includes(pageContent.title));

      // assert that page container is rendered
      const page = document.querySelector('.page');
      expect(page).toBeInTheDocument();

      // test if card compoent is rendered
      const card = document.querySelector('.MuiCard-root');
      expect(card).toBeInTheDocument();

      // assert that the component matches the existing snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe('when there is no page content', () => {
    it('it does not render the category page', () => {
      const pageContent = [] as ListPageItemFragment;

      render(<TagPage pageContent={pageContent} preview={false} />);

      // test if card compoent is not rendered
      const page = document.querySelector('.page');
      expect(page).toBeNull();
    });
  });
});
