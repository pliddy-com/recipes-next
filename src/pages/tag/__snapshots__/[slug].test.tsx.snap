// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagPage in tag/[slug].tsx when there is page content it renders the page 1`] = `
<div>
  <title>
    Site Title | Baking
  </title>
  <meta
    content="Index tag description Baking"
    name="description"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg page css-1oqqzyl-MuiContainer-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Baking
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
    >
      1 Total
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 css-wq3jbm-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
